x/fswap/keeper/proposal.go

```diff
@@ -0,0 +1,115 @@
+package keeper
+
+import (
+	"github.com/Finschia/finschia-sdk/store/prefix"
+	sdk "github.com/Finschia/finschia-sdk/types"
+	"github.com/Finschia/finschia-sdk/types/errors"
+	"github.com/Finschia/finschia-sdk/x/fswap/types"
+)
+
+func (k Keeper) MakeSwap(ctx sdk.Context, swap types.Swap) error {
+	if err := swap.ValidateBasic(); err != nil {
+		return err
+	}
+
+	isNewSwap := true
+	if _, err := k.getSwap(ctx, swap.ToDenom, swap.FromDenom); err == nil {
```

Comment:
Should we have to prohibit the reverse swaps?

Reply:
There is requirement for one-way swap only.

Reply:
`getSwap()` won't return a valid swap instance for (to, from) tuple. So until the governance explicitly add a swap for (to, from), the swap would be refused. And we can prohibit the number of supported swaps to 1.